sdk/dapp-kit/src/hooks/useSuiClientInfiniteQuery.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UseInfiniteQueryOptions } from '@tanstack/react-query';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { useSuiClientContext } from './useSuiClient.js';
+import type { SuiClient } from '@mysten/sui.js/client';
+
+interface PaginatedResult {
+	data?: unknown;
+	nextCursor?: unknown;
+	hasNextPage: boolean;
+}
+export type SuiRpcPaginatedMethodName = {
+	[K in keyof SuiClient]: SuiClient[K] extends (input: any) => Promise<PaginatedResult> ? K : never;
+}[keyof SuiClient];
+
+export type SuiRpcPaginatedMethods = {
+	[K in SuiRpcPaginatedMethodName]: SuiClient[K] extends (input: infer P) => Promise<{
+		data?: infer R;
+		nextCursor?: infer Cursor | null;
+		hasNextPage: boolean;
+	}>
+		? {
+				name: K;
+				result: R;
+				params: P;
+				cursor: Cursor;
+		  }
+		: never;
+};
+
+export type UseSuiClientInfiniteQueryOptions<T extends keyof SuiRpcPaginatedMethods> = Omit<
+	UseInfiniteQueryOptions<
+		SuiRpcPaginatedMethods[T]['result'],
+		unknown,
```

Comment:
is the error field unknown or can we default to `Error`?

Reply:
Good question, I'll have to dig into this a bit, but using Error probably makes sense